pkg/csi/service/wcp/controller.go

```diff
@@ -1278,28 +1279,24 @@ func (c *controller) ControllerUnpublishVolume(ctx context.Context, req *csi.Con
 						return nil, csifault.CSIInternalFault, logger.LogNewErrorCodef(log, codes.Internal,
 							"failed to connect to Virtual Center: %s", vc.Config.Host)
 					}
-					podVM, err := getVMByInstanceUUIDInDatacenter(ctx, vc, dcMorefValue, v)
-					if err != nil {
-						if err == cnsvsphere.ErrVMNotFound {
-							log.Infof("virtual machine not found for vmUUID %q. "+
-								"Thus, assuming the volume is detached.", v)
-							break
-						}
-						return nil, csifault.CSIInternalFault, logger.LogNewErrorCodef(log, codes.Internal,
-							"failed to get the PodVM Moref from the PodVM UUID: %s in datacenter: %s with err: %+v",
-							v, dcMorefValue, err)
-					}
 					isStillAttached := false
 					timeout := 4 * time.Minute
 					pollTime := time.Duration(5) * time.Second
-					err = wait.Poll(pollTime, timeout, func() (bool, error) {
-						podVM, err := getVMByInstanceUUIDInDatacenter(ctx, vc, dcMorefValue, v)
+					var podVM *cnsvsphere.VirtualMachine
+					err = wait.PollImmediate(pollTime, timeout, func() (bool, error) {
+						podVM, err = getVMByInstanceUUIDInDatacenter(ctx, vc, dcMorefValue, v)
 						if err != nil {
 							if err == cnsvsphere.ErrVMNotFound {
 								log.Infof("virtual machine not found for vmUUID %q. "+
 									"Thus, assuming the volume is detached.", v)
 								return true, err
```

Comment:
PollImmediate tries a condition func until it returns true, an error, or the timeout is reached.
Here, we are returning true and error. What is expected to be done? Retry condition or exit poll?

Reply:
We are switching to `PollImmediate` because we want to check Pod VM existence immediately in the loop without waiting for 5 second.
if we find Pod VM is not present we are returning true with error to break the loop.